spec/commands/spawner.spec.js

```diff
@@ -2,28 +2,104 @@
 
 const mocks = require('../mocks');
 const sut = require('../../commands/spawner');
+const Spawner = require('../../models/spawner');
+const Room = require('../../models/room');
 
 describe('spawner', function () {
   let socket;
+  let currentRoom;
 
-  beforeAll(function () {
+  beforeEach(function () {
+    currentRoom = mocks.getMockRoom();
+    currentRoom.name = 'Dance Floor';
+    currentRoom.spawner = new Spawner();
+    currentRoom.spawner.mobTypes.push(mocks.mobType.name);
     socket = new mocks.SocketMock();
+    socket.user.roomId = currentRoom.id;
+    socket.user.username = 'Disco Jim';
   });
 
   describe('execute', function () {
-    describe('when action is add', function() {
```

Comment:
It looks like these describe statements were meant to be used as a section for each command, so that multiple tests could be added for each command. Could you put the describe calls back and add "its" inside them? Typically you have tests structures that are something like that when("Some condition") it("should return false if something is something")